client/app/visualizations/gauge/index.js

```diff
@@ -0,0 +1,230 @@
+  }
+
+  function isRendered() {
+    return (svg !== undefined);
+  }
+
+  that.isRendered = isRendered;
+
+  function render(newValue) {
+    svg = d3.select(element)
+      .append('svg:svg')
+      .attr('class', 'gauge')
+      .attr('width', config.clipWidth)
+      .attr('height', config.clipHeight);
+
+    const centerTx = centerTranslation();
+
+    const arcs = svg.append('g')
+      .attr('class', 'arc')
+      .attr('transform', centerTx);
+
+    arcs.selectAll('path')
+      .data(tickData)
+      .enter().append('path')
+      .attr('fill', (d, i) => config.colors[i])
+      .attr('d', arc);
+
+    const lg = svg.append('g')
+      .attr('class', 'label')
+      .attr('transform', centerTx);
+    lg.selectAll('text')
+      .data(ticks)
+      .enter().append('text')
+      .attr('transform', (d) => {
+        const ratio = scale(d);
+        const newAngle = config.minAngle + (ratio * range);
+        return 'rotate(' + newAngle + ') translate(0,' + (config.labelInset - r) + ')';
+      })
+      .text(config.labelFormat);
+
+    const lineData = [[config.pointerWidth / 2, 0],
+      [0, -pointerHeadLength],
+      [-(config.pointerWidth / 2), 0],
+      [0, config.pointerTailLength],
+      [config.pointerWidth / 2, 0]];
+    const pointerLine = d3.svg.line().interpolate('monotone');
+    const pg = svg.append('g').data([lineData])
+      .attr('class', 'pointer')
+      .attr('transform', centerTx);
+
+    pointer = pg.append('path')
+      .attr('d', pointerLine/* function(d) { return pointerLine(d) +'Z';} */)
+      .attr('transform', 'rotate(' + config.minAngle + ')');
+
+    update(newValue === undefined ? 0 : newValue);
+  }
+
+  that.render = render;
+
+  function update(newValue, newConfiguration) {
+    if (newConfiguration !== undefined) {
+      configure(newConfiguration);
+    }
+    const ratio = scale(newValue);
+    const newAngle = config.minAngle + (ratio * range);
+    pointer.transition()
+      .duration(config.transitionMs)
+      .ease('elastic')
+      .attr('transform', 'rotate(' + newAngle + ')');
+  }
+
+  that.update = update;
+
+  configure({
+    size: 300,
+    clipWidth: 300,
+    clipHeight: 300,
+    ringWidth: 60,
+    maxValue: options.dangerRange,
+    transitionMs: 4000,
+  });
+
+  return that;
+}
+
+function isDataValid(data) {
+  // data should contain column named 'value', otherwise no reason to render anything at all
+  return _.find(data.columns, c => c.name === 'value');
+}
+
+const GaugeRenderer = {
+  template: '<div class="power-gauge" resize-event="handleResize()"></div>',
+  bindings: {
+    data: '<',
+    options: '<',
+  },
+  controller($scope, $element) {
+    const container = $element[0].querySelector('.power-gauge');
+
+    const update = () => {
+      if (this.data && isDataValid(this.data)) {
+        // do the render logic.
+        angular.element(container).empty();
+
+        createGauge(container, this.data.rows, this.options).render(this.data.rows[0].value);
```

Comment:
1. When you create something using some library - you have to destroy it using the same library. In this case probably everything will be fine (but I'm not 100% sure tho), but you create svg with `d3` - so you shouldn't use Angular's functions to destroy it.
2. You return an object from `createGauge` with a lot of functions, but later use only `render`. I think, it's better to rename it to `renderGauge`, pass all needed arguments to it and do not return anything from it at all.

Reply:
encapsulated the internal methods 
regarding destroing the d3 object I used the code from "sankey" and I don't see there something related , maybe you can reference me to relevant directions